### PR TITLE
Integrate mbedtls' CMake

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: .github/workflows/cmake-windows.yml [hunk 1]
```diff
@@ -183,7 +184,7 @@ jobs:
         endif()
 
         execute_process(
-          COMMAND ${{ steps.cmake_and_ninja.outputs.cmake_dir }}/ctest --verbose
+          COMMAND ${{ steps.cmake_and_ninja.outputs.cmake_dir }}/ctest --verbose --tests-regex "apitest|platformtest|messagingtest|securitytest"
```
[the remark this answers is not in the document]
Comment (Danielius1922) [reply .github/workflows/cmake-windows.yml:194]: These are our tests. Without filter this will also run 90+ mbedtls tests.

**Comment by Danielius1922** [comment]
@SiMet, could you please check the OCF Conformance job, it fails at the start and doesn't run. Thanks.

File: .github/workflows/cmake-windows.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Danielius1922) [reply .github/workflows/cmake-windows.yml]: I was able to use `label` property, which will be automatically added by cmake to our tests and then I can run tests only with that label.

File: CMakeLists.txt [hunk 2]
```diff
@@ -83,11 +83,14 @@ set(OC_CLANG_TIDY_ENABLED OFF CACHE BOOL "Enable clang-tidy analysis during comp
 
 include(tools/clang-tidy.cmake)
 
-######## Define compiler flags ########
+# ####### Define compiler flags ########
```
Comment (Danielius1922) [inline CMakeLists.txt:86]: I wish the cmake auto-formatter didn't do this, oh well. :D

**Comment by Danielius1922** [comment]
> > @SiMet, could you please check the OCF Conformance job, it fails at the start and doesn't run. Thanks.
> 
> I've just look at this and probably fix.

It got further, but still failed (previously it failed after 30s, now it lasted 6+mins).

**Comment by SiMet** [comment]
> > > @SiMet, could you please check the OCF Conformance job, it fails at the start and doesn't run. Thanks.
> > 
> > 
> > I've just look at this and probably fix.
> 
> It got further, but still failed (previously it failed after 30s, now it lasted 6+mins).

@Danielius1922 it finally passed :)

**Review by jkralik — APPROVED** [review]

**Review by jkralik — APPROVED** [review]

File: .github/workflows/cmake-linux-sanitized.yml [hunk 3]
```diff
@@ -39,10 +39,10 @@ jobs:
         run: |
           mkdir linuxbuild
           cd linuxbuild
-          cmake ${{ matrix.args }} -DCMAKE_BUILD_TYPE=Debug -DOC_CLOUD_ENABLED=ON -DOC_COLLECTIONS_IF_CREATE_ENABLED=ON -DOC_IPV4_ENABLED=ON -DBUILD_TESTING=ON ../.
+          cmake ${{ matrix.args }} -DUSE_SHARED_MBEDTLS_LIBRARY=ON -DCMAKE_BUILD_TYPE=Debug -DOC_CLOUD_ENABLED=ON -DOC_COLLECTIONS_IF_CREATE_ENABLED=ON -DOC_IPV4_ENABLED=ON -DBUILD_TESTING=ON ../.
```
Comment (Danielius1922) [inline .github/workflows/cmake-linux-sanitized.yml:42]: TODO: rm this was just to test if this works

**Pull request merged**